src/controller.js

```diff
@@ -48,18 +48,33 @@ function getKernelSource() {
   return `(${kernelSourceFunc})`;
 }
 
-// this feeds the SES realm's (real/safe) confineExpr() back into the Realm
+// this feeds the SES realm's (real/safe) makeCompartment() back into the Realm
 // when it does require('@agoric/evaluate'), so we can get the same
 // functionality both with and without SES
 function makeEvaluate(e) {
-  const { confineExpr, confine } = e;
-  const evaluateExpr = (source, endowments = {}, options = {}) =>
-    confineExpr(source, endowments, options);
-  const evaluateProgram = (source, endowments = {}, options = {}) =>
-    confine(source, endowments, options);
-  return Object.assign(evaluateExpr, {
-    evaluateExpr,
-    evaluateProgram,
+  const { makeCompartment, sesOptions } = e;
+  const { shims, transforms } = sesOptions;
+  const makeEvaluators = (realmOptions = {}) => {
+    const { shims: realmShims, transforms: realmTransforms } = realmOptions;
+    const c = makeCompartment({
+      ...sesOptions,
+      ...realmOptions,
+      shims: (shims || []).concat(realmShims || []),
```

Comment:
This might be clearer without lines 56 and 58. So:

```
shims: (sesOptions.shims || []).concat(realmOptions.shims || [])
transforms: (realmOptions.transforms || []).concat(shimOptions.transforms || [])
```

also maybe a comment about why they're in opposite orders (I'm guessing the later shims are more powerful than the earlier ones, but the later transforms are less powerful, or the other way around)

Reply:
Done.